tests/regression/test_interpolate.py

```diff
@@ -0,0 +1,121 @@
+import pytest
+import numpy as np
+from firedrake import *
+from tests.common import *
+
+
+def test_constant(cg1):
+    f = interpolate(Constant(1.0), cg1)
+    assert np.allclose(1.0, f.dat.data)
+
+
+def test_function():
+    m = UnitTriangleMesh()
+    V1 = FunctionSpace(m, 'P', 1)
+    V2 = FunctionSpace(m, 'P', 2)
+
+    f = interpolate(Expression("x[0]*x[0]"), V1)
+    g = interpolate(f, V2)
+
+    # g shall be equivalent to:
+    h = interpolate(Expression("x[0]"), V2)
+
+    assert np.allclose(g.dat.data, h.dat.data)
+
+
+def test_coordinates(cg2):
+    f = interpolate(Expression("x[0]*x[0]"), cg2)
+
+    x = cg2.mesh().coordinates
+    g = interpolate(x[0]*x[0], cg2)
+
+    assert np.allclose(f.dat.data, g.dat.data)
+
+
+def test_piola():
+    m = UnitTriangleMesh()
+    U = FunctionSpace(m, 'RT', 1)
+    V = FunctionSpace(m, 'P', 2)
+
+    f = project(Expression(("x[0]", "0.0")), U)
+    g = interpolate(f[0], V)
+
+    # g shall be equivalent to:
+    h = project(f[0], V)
+
+    assert np.allclose(g.dat.data, h.dat.data)
+
+
+def test_vector():
+    m = UnitTriangleMesh()
+    U = FunctionSpace(m, 'RT', 1)
+    V = VectorFunctionSpace(m, 'P', 2)
+
+    f = project(Expression(("x[0]", "0.0")), U)
+    g = interpolate(f, V)
+
+    # g shall be equivalent to:
+    h = project(f, V)
+
+    assert np.allclose(g.dat.data, h.dat.data)
+
+
+def test_constant_expression():
+    m = UnitTriangleMesh()
+    U = FunctionSpace(m, 'RT', 1)
+    V = FunctionSpace(m, 'P', 2)
+
+    f = project(Expression(("x[0]", "x[1]")), U)
+    g = interpolate(div(f), V)
+
+    assert np.allclose(2.0, g.dat.data)
+
+
+def test_compound_expression():
+    m = UnitTriangleMesh()
+    x = m.coordinates
+    U = FunctionSpace(m, 'RT', 2)
+    V = FunctionSpace(m, 'P', 2)
+
+    f = project(Expression(("x[0]", "x[1]")), U)
+    g = interpolate(Constant(1.5)*div(f) + sin(x[0] * np.pi), V)
+
+    # g shall be equivalent to:
+    h = interpolate(Expression("3.0 + sin(pi * x[0])"), V)
+
+    assert np.allclose(g.dat.data, h.dat.data)
+
+
+def test_cell_orientation():
+    m = UnitCubedSphereMesh(1)
+    m.init_cell_orientations(Expression(('x[0]', 'x[1]', 'x[2]')))
+    x = m.coordinates
+    U = FunctionSpace(m, 'RTCF', 1)
+    V = VectorFunctionSpace(m, 'Q', 1)
+
+    f = project(as_tensor([x[1], -x[0], 0.0]), U)
+    g = interpolate(f, V)
+
+    # g shall be equivalent to:
+    h = interpolate(f, V)
```

Comment:
So here you're checking that g is equivalent to g right?  Because you're just interpolating f twice.